src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultCodeRenderingContext.cs

```diff
@@ -81,6 +80,10 @@ internal class DefaultCodeRenderingContext : CodeRenderingContext
         // This will be initialized by the document writer when the context is 'live'.
         public IntermediateNodeVisitor Visitor { get; set; }
 
+        public override IEnumerable<IntermediateNode> Ancestors => _ancestors;
+
+        internal Stack<IntermediateNode> AncestorsInternal => _ancestors;
```

Comment:
You already have a test code rendering context, expose this there, overkill to do this in product code.